### DISCUSSION TITLE
Statically linked OpenSSL

### DISCUSSION THREAD

**Discussion by theAkito** [nushell/nushell#7280] [Q&A]
With the release of [0.72.0](https://github.com/nushell/nushell/releases/tag/0.72.0) I have noticed, that OpenSSL is being statically linked.

I've searched issues, discussions & pull requests, but couldn't find a discussion about how that decision came to be & whether alternatives were considered. (https://github.com/nushell/nushell/pull/5349 https://github.com/nushell/nushell/pull/5352)

Was choosing [OpenSSL](https://www.openssl.org/) over an alternative, like for example [LibreSSL](https://www.libressl.org/), a conscious decision?

**Comment by kubouch** [comment]
IIRC there were issues when people had a wrong / incompatible version of OpenSSL installed or something. It was causing a lot of troubles when it was linked dynamically.

As for the LibreSSL, I have no clue, but maybe @rgwood might know.

**Reply by rgwood** [reply]
Yeah it's been statically linked in release binaries since April. The motivation was basically that many distros ship with OpenSSL 3, and if you ship a binary dynamically linked against v3 it will have problems on a distro using v1 (and vice versa).

I don't think much thought went into using OpenSSL specifically, it was just the default for a lot of our dependencies.

We may want to revisit using rustls; it solves the same problem. I believe we tried it briefly but dropped it because someone had trouble building it on PowerPC. Not sure if that limitation still applies, but IMO PowerPC is esoteric enough that it shouldn't be a blocker https://github.com/nushell/nushell/pull/3120
